presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+    public TaskInfo getTaskInfo()
+    {
+        return taskInfo.get();
+    }
+
+    public void start()
+    {
+        sendUpdateRequest();
+        scheduleGetTaskInfo();
+    }
+
+    public void stop()
+    {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+        }
+    }
+
+    private void sendUpdateRequest()
+    {
+        Optional<byte[]> fragment = Optional.of(planFragment.toBytes(planFragmentCodec));
+        Optional<TableWriteInfo> writeInfo = Optional.of(tableWriteInfo);
+        TaskUpdateRequest updateRequest = new TaskUpdateRequest(
+                session.toSessionRepresentation(),
+                session.getIdentity().getExtraCredentials(),
+                fragment,
+                sources,
+                outputBuffers,
+                writeInfo);
+        byte[] taskUpdateRequestJson = taskUpdateRequestCodec.toBytes(updateRequest);
+
+        HttpUriBuilder uriBuilder = uriBuilderFrom(getTaskInfo().getTaskStatus().getSelf());
+        Request request = setContentTypeHeaders(false, preparePost())
+                .setUri(uriBuilder.build())
+                .setBodyGenerator(createStaticBodyGenerator(taskUpdateRequestJson))
+                .build();
+
+        ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+        ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+        FutureCallback callback = new SimpleHttpResponseHandler<>(new UpdateResponseHandler(), request.getUri(), stats.getHttpResponseStats(), NATIVE_EXECUTION_TASK_ERROR);
+
+        Futures.addCallback(
+                future,
+                callback,
+                executor);
+    }
+
+    private void scheduleGetTaskInfo()
+    {
+        scheduledFuture = updateScheduledExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                HttpUriBuilder uriBuilder = uriBuilderFrom(getTaskInfo().getTaskStatus().getSelf());
+                Request request = setContentTypeHeaders(false, prepareGet())
+                        .setUri(uriBuilder.build())
+                        .build();
+
+                ResponseHandler responseHandler = createAdaptingJsonResponseHandler(taskInfoCodec);
+                ListenableFuture<BaseResponse<TaskInfo>> future = httpClient.executeAsync(request, responseHandler);
+                FutureCallback callback = new SimpleHttpResponseHandler<>(new TaskInfoHandler(), request.getUri(), stats.getHttpResponseStats(), NATIVE_EXECUTION_TASK_ERROR);
+
+                Futures.addCallback(
+                        future,
+                        callback,
+                        executor);
+            }
+            catch (Throwable t) {
+                throw t;
+            }
+        }, 0, GET_TASK_INFO_INTERVALS, MILLISECONDS);
+    }
+
+    private void updateTaskInfo(TaskInfo newValue)
+    {
+        taskInfo.set(newValue);
```

Comment:
can inline this method now if you want since it's only one line and doesn't need any special synchronization or anything.